features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/thresholding/FlowThresholding.java

```diff
@@ -76,36 +82,20 @@
 
     private final IpInterfaceDao ipInterfaceDao;
 
-    private final Map<ApplicationKey, AtomicLong> applicationAccumulator;
-
-    private final Cache<ExporterKey, Session> sessions;
+    private final ConcurrentMap<ExporterKey, Session> sessions = Maps.newConcurrentMap();
 
     private long stepSizeMs = 0;
+    private long idleTimeoutMs = 15 * 60 * 1000;
```

Comment:
Should be volatile. Otherwise timer task may not see the current value.

Reply:
AFAIK `volatile` is only required for multiple access of the same variable in the same method. Method boundaries are memory barriers and invalidate caches anyways. This is not a tight spinning loop.

Even if this need does a run with the old value after changing the config, this is fine.

Reply:
Method boundaries have nothing to do with memory barriers. Volatile guaranties that changes made by a thread to a volatile variable are seen other threads accessing that variable.

Reply:
altered.